### PR TITLE
Ruby/Hamming: Add pure String solution

### PR DESCRIPTION
**Pull request by F3PiX**
- Adds a solution that doesn't convert to arrays.
- Rephrased a bit.

### CODE REVIEW

**Comment by F3PiX** [comment]
@JoelQ I rephrased some of your texts to make room for a second perfect ;-) solution. Can you check please? 
Also if you think this String only solution deserves this prominent place.

**Comment by JoelQ** [comment]
👍for moving away from the first-person language.

The guide details three strategies for iteration: 

1. Iterating through both strings simultaneously via `Enumerable#zip` (avoids
   the need for indices)
2. Iterating through indices directly using a `Range`, `Integer#upto`, or
   `Integer#times`.
3. A hybrid approach where you iterate through one of the strings while also
   tracking the index to find the equivalent character in the other string. Uses
`Enumerable#each_with_index`.

I originally had examples for each but ended up cutting them to make the guide shorter. Do you think we should have examples for each?

Since we already have a solution for approach 1, you're adding a solution for approach 2 in this PR, maybe it makes sense to also have one for approach 3?

**Comment by F3PiX** [comment]
> Do you think we should have examples for each?
Hmm, yeah, I like that. It would make the mentoring easier if the solution is available in the notes
I'll add them.

About this solution being an example of the second strategy: no doubt it is. What sets it apart, is that it is working with the strings, not converting them to arrays. 
For an example to the second strategy, I'd rather add a - more common - array example. 
And leave the string-y one as a separate example. 
What do you think?

**Comment by JoelQ** [comment]
I think all strategy 2 solutions are string-based?

```ruby
# Range
(0..s1.length).count { |i| s1[i] != s2[i] }

# Integer#upto
0.upto(s1.length).count { |i| s1[i] != s2[i] }

# Integer#times
s1.length.times.count { |i| s1[i] != s2[i] }
```

That's probably something important to highlight! Maybe mention some strengths and weaknesses of the various strategies?

1. Iterating via `zip` (👍 avoids indices entirely, 👎 needs to convert the string to an array)
2. Iterating explicit indices via `Range`/`upto`/`times` (👍 uses string directly, no array conversion needed, 👎 requires indices)
3. Hybrid approach with `each_with_index` (👍 more intuitive? students seem to pick this often, 👎 requires both indices _and_ converting to arrays so it's kind of the worst of both worlds from 1 and 2 😛 )

**Comment by F3PiX** [comment]
O, of course! I totally overlooked that all strategy-2 solutions are strings only. Which confirms that we should highlight it :-) 
I'll add it, thanks.

File: tracks/ruby/exercises/hamming/README.md [hunk 1]
```diff
@@ -36,24 +43,21 @@ class Hamming
...
+```ruby 
+    (0..strand1.length).count { |i| strand1[i] != strand2[i] }
+```
 
-translate almost directly into the Ruby code:
+Strategy 3: Each with index
 
 ```ruby
-.count { |char1, char2| char1 != char2 }
+    some_strand.chars.each_with_index.count {|letter, index| letter != other_strand[index] }
```
Comment (JoelQ) [inline tracks/ruby/exercises/hamming/README.md]: Can we use the variable names `strand1` and `strand2` so the examples remain consistent?

File: tracks/ruby/exercises/hamming/README.md [hunk 2]
```diff
@@ -1,30 +1,37 @@
...
-3. A hybrid approach where you iterate through one of the strings while also
-   tracking the index to find the equivalent character in the other string. Uses
-`Enumerable#each_with_index`.
+There are three strategies for the **iteration**. In descending order of popularity with mentors:
+ 
+Hybrid approach 
+1. Iterating through both strings simultaneously via `Enumerable#zip`: 👍 avoids indices entirely, 👎 needs to convert the string to an array
+2. Iterating explicit indices using a `Range`, `Integer#upto`, or `Integer#times`: 👍 uses string directly, no array conversion needed, 👎 requires indices
+3. A hybrid approach with `Enumerable#each_with_index`, iterating over one of the strings while also
+   tracking the index to find the equivalent character in the other string. Uses `Enumerable#each_with_index`: 👍 more intuitive? students seem to pick this often, 👎 requires both indices and converting to arrays so it's kind of the worst of both worlds from 1 and 2 😛
```
Comment (JoelQ) [inline tracks/ruby/exercises/hamming/README.md]: I wonder if we're packing a little too much information in this ordered list?  Would it make sense to move the analysis of pros/cons to a paragraph above each example? Or do you think it's better to keep it all together in this section here?

File: tracks/ruby/exercises/hamming/README.md (hunk 1 above)
Comment (F3PiX) [reply tracks/ruby/exercises/hamming/README.md]: I liked that is was more evidently different solutions. But apparently it's not more clear ;-) 
I'll change it. Thanks!

File: tracks/ruby/exercises/hamming/README.md (hunk 2 above)
Comment (F3PiX) [reply tracks/ruby/exercises/hamming/README.md]: Maybe we do pack too much here. Let me try something else....

**Comment by F3PiX** [comment]
Looks better to me, @JoelQ, thanks!!

**Comment by JoelQ** [comment]
Looks good!

**Pull request merged**